src/mame/drivers/1945kiii.cpp

```diff
@@ -281,6 +297,25 @@ u32 k3_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const r
 	return 0;
 }
 
+void k3_state::vram_buffer(bool newval)
+{
+	const bool oldval = m_refresh;
+	if (!oldval && newval)
+	{
+		for (int j = 0; j < 2; j++)
+		{
+			const u32 size = m_spriteram[j].bytes() / 2;
+			for (int i = 0; i < size; i++)
+				m_spriteram_buffer[j][i] = m_spriteram[j][i];
+		}
+		const u32 size = m_bgram.bytes() / 2;
+		for (int i = 0; i < size; i++)
+			m_bgram_buffer[i] = m_bgram[i];
+
+		m_bg_tilemap->mark_all_dirty();
+	}
+	m_refresh = newval;
+}
```

Comment:
Wouldn't it be a better representation to flip pointers rather than doing copies like this?